dcpy/utils/geospatial.py

```diff
@@ -22,6 +40,72 @@
 }
 
 
+def convert_to_geodata(
+    data: pd.DataFrame,
+    geometry_format: str,
+    geometry_column: str,
+    new_geometry_column: str = "geometry",
+) -> gpd.GeoDataFrame:
+    data = data.copy()
+    match geometry_format:
+        case "WKT":
+            data[geometry_column] = gpd.GeoSeries.from_wkt(data[geometry_column])
+        case "WKB":
+
+            def _try_wkb(wkb):
+                try:
+                    return pd.Series([shapely.from_wkb(wkb), None])
+                except Exception as e:
+                    return pd.Series([None, str(e)])
+
+            data[[new_geometry_column, "geometry_error"]] = data.apply(
+                lambda row: _try_wkb(row[geometry_column]), axis=1
+            )
+        case _:
+            raise NotImplementedError(
+                f"Unsupported geometry format {geometry_format:}. Must be one of the keys in the dict {GEOMETRY_FORMATS:}"
+            )
+    data.drop([geometry_column], axis=1, inplace=True)
+    geo_data = gpd.GeoDataFrame(
+        data,
+        geometry=new_geometry_column,
+        crs=PROJECTIONS["WGS_84_DEG"],
```

Comment:
Just not familiar with gdf constructor - is this assuming that the input is in wgs84? Or converting to

Reply:
unfortunately this is assuming that it's wgs84. to do spatial things later like reprojecting via `GeoDataFrame.to_crs`, it has to have a crs set in the first place, so I figured might as well do it when it's created

would probably be better to make the CRS an argument to `convert_to_geodata` and not assume it here?